Firestore/Example/Tests/Integration/API/FIRDatabaseTests.mm

```diff
@@ -1744,4 +1753,37 @@ - (void)testWaitForPendingWritesCompletesWhenOfflineIfNoPending {
   [self awaitExpectations];
 }
 
+- (void)testDefaultNamedDbIsSame {
+  [FIRApp configure];
+  FIRApp *app = [FIRApp defaultApp];
+  FIRFirestore *db1 = [FIRFirestore firestore];
+  FIRFirestore *db2 = [FIRFirestore firestoreForApp:app];
+  FIRFirestore *db3 = [FIRFirestore firestoreForApp:app database:@"(default)"];
+  FIRFirestore *db4 = [FIRFirestore firestoreForDatabase:@"(default)"];
+
+  XCTAssertIdentical(db1, db2);
+  XCTAssertIdentical(db1, db3);
+  XCTAssertIdentical(db1, db4);
+}
+
+- (void)testSameNamedDbIsSame {
+  [FIRApp configure];
+  FIRApp *app = [FIRApp defaultApp];
+  FIRFirestore *db1 = [FIRFirestore firestoreForApp:app database:@"myDb"];
+  FIRFirestore *db2 = [FIRFirestore firestoreForDatabase:@"myDb"];
+
+  XCTAssertIdentical(db1, db2);
+}
+
+- (void)testNamedDbHaveDifferentPersistence {
```

Comment:
FWIW, here it is testing if they are the same instance, not the same persistence underlying..maybe this is a typo?

Reply:
Persistence -> Instance ?